### DISCUSSION TITLE
Where's toolchain_configure?

### DISCUSSION THREAD

**Discussion by grepwood** [bazel-contrib/rules_oci#684] [Q&A] [answered]
I'm one of those users who use a pre-installed Docker toolchain, as described here: https://github.com/bazelbuild/rules_docker/blob/3040e1fd74659a52d1cdaff81359f57ee0e2bb41/README.md?plain=1#L127

My setup relies on a certain vendor injecting the Docker version they see fit in their product. I would like to not stray away from their solution, because when they find out that my Bazel code tries to run its own show with its own Docker binary, they may try to discount all of their responsibility if something tangentially related to Docker doesn't work in my setup, and shift the responsibility for this on me.

Where can I find the settings to configure the Docker toolchain in rules_oci or anything else that will completely eliminate this behavior?
https://github.com/bazel-contrib/rules_oci/blob/bf4c60bc8c9b4ff242d4c353cf8a3e1bf3284fe6/README.md?plain=1#L27

**Comment by thesayyn** [comment]
rules_oci does not have docker dependency, therefore does need a docker toolchain.

**Reply by grepwood** [reply]
I'm reading https://docs.aspect.build/guides/rules_oci_migration/ and I see that a lot of functions are already covered by rules_oci. How come rules_oci skips the requirement for an adequate toolchain?

**Reply by thesayyn** [reply, marked as answer]
I am not sure where the requirements come from, nowhere in rules_oci we depend on `docker` executable, everything we depend on is brought by a hermetic toolchain, eg: crane, regctl. 

With the expection of oci_load; 

originally oci_load did not depend on the host docker client, we just told people to use `bazel build :tarball` and do `docker load -i bazel-bin/pth/tarball.tar` 

However people wanted a convenient way to do `bazel build :oci_tarball | docker load` so we made oci_load runnable which does exactly this.

**Reply by loeffel-io** [reply]
but how to use crane in v2.0.1? @thesayyn

**Reply by thesayyn** [reply]
We didn't change anything about crane. it's same as before.
